frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/hooks.ts

```diff
@@ -262,3 +262,29 @@ export const useNodes = (
 
   return [...nodes, finallyNode];
 };
+
+const touchTaskFields = (task: PipelineTask): FormikTouched<PipelineTask> => ({
+  workspaces: task.workspaces?.map(() => ({ workspace: true })),
+  resources: {
+    inputs: task.resources?.inputs?.map(() => ({ resource: true })),
+    outputs: task.resources?.outputs?.map(() => ({ resource: true })),
+  },
+});
+
+export const useExplicitFormikTouch = () => {
```

Comment:
nit: This name `useExplicitFormikTouch` is generic but the implementation seems its tied to `PipelineTask`'s workspace and resources, so I am thinking if we could change it to something that is builder specific.